agent/exec/container/controller.go

```diff
@@ -194,6 +195,12 @@ func (r *controller) Wait(pctx context.Context) error {
 				// If we get here, something has gone wrong but we want to exit
 				// and report anyways.
 				return ErrContainerDestroyed
+			case "health_status: unhealthy":
```

Comment:
@nishanttotla yes, I tested it by running the demo I showed you yesterday.  In that demo, container will keep running, but health check would fail. And `health_status: unhealthy` shows up as an event.
I will double check it by running more end-to-end tests. It's a little hard to implement health check tests in gmock.